extruct/dublincore.py

```diff
@@ -0,0 +1,157 @@
+    'haspart': 'http://purl.org/dc/terms/hasPart',
+    'hasversion': 'http://purl.org/dc/terms/hasVersion',
+    'instructionalmethod': 'http://purl.org/dc/terms/instructionalMethod',
+    'methodofinstruction': 'http://purl.org/dc/terms/MethodOfInstruction',
+    'isformatof': 'http://purl.org/dc/terms/isFormatOf',
+    'ispartof': 'http://purl.org/dc/terms/isPartOf',
+    'isreferencedby': 'http://purl.org/dc/terms/isReferencedBy',
+    'isreplacedby': 'http://purl.org/dc/terms/isReplacedBy',
+    'isrequiredby': 'http://purl.org/dc/terms/isRequiredBy',
+    'issued': 'http://purl.org/dc/terms/issued',
+    'isversionof': 'http://purl.org/dc/terms/isVersionOf',
+    'language': 'http://purl.org/dc/terms/language',
+    'linguisticsystem': 'http://purl.org/dc/terms/LinguisticSystem',
+    'license': 'http://purl.org/dc/terms/license',
+    'licensedocument': 'http://purl.org/dc/terms/LicenseDocument',
+    'mediator': 'http://purl.org/dc/terms/mediator',
+    'medium': 'http://purl.org/dc/terms/medium',
+    'physicalresource': 'http://purl.org/dc/terms/PhysicalResource',
+    'physicalmedium': 'http://purl.org/dc/terms/PhysicalMedium',
+    'modified': 'http://purl.org/dc/terms/modified',
+    'provenance': 'http://purl.org/dc/terms/provenance',
+    'provenancestatement': 'http://purl.org/dc/terms/ProvenanceStatement',
+    'publisher': 'http://purl.org/dc/terms/publisher',
+    'references': 'http://purl.org/dc/terms/references',
+    'replaces': 'http://purl.org/dc/terms/replaces',
+    'requires': 'http://purl.org/dc/terms/requires',
+    'rightsholder': 'http://purl.org/dc/terms/rightsHolder',
+    'source': 'http://purl.org/dc/terms/source',
+    'spatial': 'http://purl.org/dc/terms/spatial',
+    'location': 'http://purl.org/dc/terms/Location',
+    'subject': 'http://purl.org/dc/terms/subject',
+    'tableofcontents': 'http://purl.org/dc/terms/tableOfContents',
+    'temporal': 'http://purl.org/dc/terms/temporal',
+    'periodoftime': 'http://purl.org/dc/terms/PeriodOfTime',
+    'type': 'http://purl.org/dc/terms/type',
+    'valid': 'http://purl.org/dc/terms/valid',
+}
+
+_URL_NAMESPACES = ['http://purl.org/dc/terms/', 'http://purl.org/dc/elements/1.1/']
+
+
+def get_lower_attrib(name):
+    # get attribute to compare against _DC_TERMS or _DC_ELEMENTS
+    return re.sub(r".*\.", "", name).lower()
+
+
+class DublinCoreExtractor(object):
+    """DublinCore extractor following extruct API."""
+
+    def extract(self, htmlstring, base_url=None, encoding='UTF-8'):
+        tree = parse_html(htmlstring, encoding=encoding)
+        return list(self.extract_items(tree, base_url=base_url))
+
+    def extract_items(self, document, base_url=None):
+        elements = []
+        terms = []
+
+        def attrib_to_dict(attribs):
+            # convert _attrib type to dict
+            node_dict = {}
+            for attrib, value in attribs.items():
+                node_dict.update({attrib: value})
+            return node_dict
```

Comment:
Is it different from `return dict(attribs)`, or maybe `return dict(attribs.items()`?

Reply:
Yes you're right :muscle:, thanks for feedback :pray: